src/main/transports/net.ts

```diff
@@ -117,8 +119,11 @@ export class NetTransport extends Transports.BaseTransport {
             // Drain
           });
         });
-        req.write(request.body);
-        req.end();
+
+        gzip(request.body, (_, compressedBody) => {
+          req.write(compressedBody);
```

Comment:
This is fine. If you like to reduce overheads here, you could also create a gzip stream and then have it flush into the request, like described [here](https://nodejs.org/api/zlib.html#zlib_flushing).